fontforgeexe/cvpalettes.c

```diff
@@ -819,100 +817,281 @@ static void CVPolyStar(CharView *cv) {
+	{ &GIcon_corner_selected,   &GIcon_tangent_selected},
+	{ &GIcon_scale_selected,    &GIcon_rotate_selected },
+	{ &GIcon_flip_selected,     &GIcon_skew_selected },
+	{ &GIcon_3drotate_selected, &GIcon_perspective_selected },
+	{ &GIcon_rect_selected,     &GIcon_poly_selected},
+	{ 0, 0 },
+	{ &GIcon_elipse_selected,   &GIcon_star_selected},
+	{ 0, 0 },
+	{ 0, 0 }
+    }};
+static GImage *spirobuttons[2][14][2] = {
+    {
+	{ &GIcon_pointer,      &GIcon_magnify },
+	{ &GIcon_freehand,     &GIcon_hand },
+	{ &GIcon_knife,        &GIcon_ruler },
+	{ &GIcon_spiroright,   &GIcon_spirodown },
+	{ &GIcon_spirocurve,   &GIcon_spirog2curve },
+	{ &GIcon_spirocorner,  &GIcon_spiroleft },
+	{ &GIcon_scale,        &GIcon_rotate },
+	{ &GIcon_flip,         &GIcon_skew },
+	{ &GIcon_3drotate,     &GIcon_perspective },
+	{ &GIcon_rect,         &GIcon_poly},
+	{ 0, 0 },
+	{ &GIcon_elipse,       &GIcon_star},
+	{ 0, 0 },
+	{ 0, 0 }
+    }
+    , {
+	{ &GIcon_pointer_selected,     &GIcon_magnify_selected },
+	{ &GIcon_freehand_selected,    &GIcon_hand_selected },
+	{ &GIcon_knife_selected,       &GIcon_ruler_selected },
+	{ &GIcon_spiroright_selected,  &GIcon_spirodown_selected },
+	{ &GIcon_spirocurve_selected,  &GIcon_spirog2curve_selected },
+	{ &GIcon_spirocorner_selected, &GIcon_spiroleft_selected },
+	{ &GIcon_scale_selected,       &GIcon_rotate_selected },
+	{ &GIcon_flip_selected,        &GIcon_skew_selected },
+	{ &GIcon_3drotate_selected,    &GIcon_perspective_selected },
+	{ &GIcon_rect_selected,        &GIcon_poly_selected},
+	{ 0, 0 },
+	{ &GIcon_elipse_selected,      &GIcon_star_selected},
+	{ 0, 0 },
+	{ 0, 0 }
+    }};
+static GImage *normsmalls[] = { &GIcon_smallpointer,  &GIcon_smallmag,
+				&GIcon_smallpencil,   &GIcon_smallhand,
+				&GIcon_smallknife,    &GIcon_smallruler,
+				&GIcon_smallpen,      NULL,
+				&GIcon_smallcurve,    &GIcon_smallhvcurve,
+				&GIcon_smallcorner,   &GIcon_smalltangent,
+				&GIcon_smallscale,    &GIcon_smallrotate,
+				&GIcon_smallflip,     &GIcon_smallskew,
+				&GIcon_small3drotate, &GIcon_smallperspective,
+				&GIcon_smallrect,     &GIcon_smallpoly,
+				&GIcon_smallelipse,   &GIcon_smallstar };
+static GImage *spirosmalls[] = { &GIcon_smallpointer, &GIcon_smallmag,
+				 &GIcon_smallpencil,  &GIcon_smallhand,
+				 &GIcon_smallknife,   &GIcon_smallruler,
+				 &GIcon_smallspiroright,  NULL,
+				 &GIcon_smallspirocurve,  &GIcon_smallspirog2curve,
+				 &GIcon_smallspirocorner, &GIcon_smallspiroleft,
+				 &GIcon_smallscale,       &GIcon_smallrotate,
+				 &GIcon_smallflip,        &GIcon_smallskew,
+				 &GIcon_small3drotate,    &GIcon_smallperspective,
+				 &GIcon_smallrect,        &GIcon_smallpoly,
+				 &GIcon_smallelipse,      &GIcon_smallstar };
+
+static int getSmallIconsHeight()
+{
+    return GIcon_smallpointer.u.image->height;
+}
+
+static int getToolbarWidth( CharView *cv ) 
+{
+    int cache = 0;
+    if( !cache ) {
+	GImage* (*buttons)[14][2] = (CVInSpiro(cv) ? spirobuttons : normbuttons);
```

Comment:
I don't understand this declaration. What does it mean? Is it a 14 by 2 array of `GImage**`?